homeassistant/components/homekit/type_fans.py

```diff
@@ -54,28 +55,25 @@ class Fan(HomeAccessory):
     Currently supports: state, speed, oscillate, direction.
     """
 
-    def __init__(self, *args):
+    def __init__(self, *args: Any, category: str = CATEGORY_FAN) -> None:
         """Initialize a new Fan accessory object."""
-        super().__init__(*args, category=CATEGORY_FAN)
+        super().__init__(*args, category=category)
         self.chars = []
         state = self.hass.states.get(self.entity_id)
 
-        features = state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
-        percentage_step = state.attributes.get(ATTR_PERCENTAGE_STEP, 1)
-        self.preset_modes = state.attributes.get(ATTR_PRESET_MODES)
+        if state is not None:
```

Comment:
Maybe I miss something: why this check ?

Reply:
Good question. Without this check, it complained that state was optional (type | None) so it couldn’t resolve the references. That could be because I added types into the constructor (guessing, I’m not a Python expert).